data-prepper-api/src/main/java/org/opensearch/dataprepper/model/codec/JsonDecoder.java

```diff
@@ -17,32 +17,81 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 import java.util.function.Consumer;
 
 public class JsonDecoder implements ByteDecoder {
     private final ObjectMapper objectMapper = new ObjectMapper();
     private final JsonFactory jsonFactory = new JsonFactory();
+    private String keyName;
+    private Collection<String> includeKeys;
+    private Collection<String> includeKeysMetadata;
+
+    public JsonDecoder(String keyName, Collection<String> includeKeys, Collection<String> includeKeysMetadata) {
+        this.keyName = keyName;
+        this.includeKeys = includeKeys;
+        this.includeKeysMetadata = includeKeysMetadata;
+    }
+
+    public JsonDecoder() {
+        this.keyName = null;
+        this.includeKeys = null;
+        this.includeKeysMetadata = null;
+    }
 
     public void parse(InputStream inputStream, Instant timeReceived, Consumer<Record<Event>> eventConsumer) throws IOException {
         Objects.requireNonNull(inputStream);
         Objects.requireNonNull(eventConsumer);
 
         final JsonParser jsonParser = jsonFactory.createParser(inputStream);
 
+        Map<String, Object> includeKeysMap = new HashMap<>();
+        Map<String, Object> includeMetadataKeysMap = new HashMap<>();
         while (!jsonParser.isClosed() && jsonParser.nextToken() != JsonToken.END_OBJECT) {
+            final String nodeName = jsonParser.currentName();
+
+            if (includeKeys != null && includeKeys.contains(nodeName) ||
+                    (includeKeysMetadata != null && includeKeysMetadata.contains(nodeName))) {
+                jsonParser.nextToken();
+                if (includeKeys != null && includeKeys.contains(nodeName)) {
+                    includeKeysMap.put(nodeName, jsonParser.getValueAsString());
+                }
+                if (includeKeysMetadata != null && includeKeysMetadata.contains(nodeName)) {
+                    includeMetadataKeysMap.put(nodeName, jsonParser.getValueAsString());
+                }
+                continue;
+            }
+
             if (jsonParser.getCurrentToken() == JsonToken.START_ARRAY) {
-                parseRecordsArray(jsonParser, timeReceived, eventConsumer);
+                if (keyName != null && !nodeName.equals(keyName)) {
```

Comment:
This only allows for first level of fields? Should we make it generic to allow nested fields, like
```
{
   "key1" : "value1",
   "key2" : {
             "logEvents" : [
             ]
    }
}
```